enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -8569,14 +8638,20 @@ class AdjointGenerator
           getForwardBuilder(Builder2);
           auto origfree = orig->getArgOperand(0);
           auto tofree = gutils->invertPointerM(origfree, Builder2);
+
           if (tofree != origfree) {
```

Comment:
Okay I thought about this even more and the reason this is unsafe for regular AD makes this incredibly danger for split. Specifically suppose we allocate memory for one reverse, which we then splat into a width number vector. We would end up calling free on that same one width number of times. I do really think we need to add a runtime check.

Also as a note the origfree should be getNewFromOriginal